tests/test_benchmarking/test_data/test_simulated_data/test_mock_student_provider.py

```diff
@@ -21,6 +21,76 @@ def test_get__returns_student_objects(self):
         for student in students:
             self.assertIsInstance(student, Student)
 
+    def test_get__each_student_has_correct_num_friends_and_enemies(self):
+        students = MockStudentProvider(
+            MockStudentProviderSettings(
+                number_of_students=10,
+                number_of_friends=4,
+                number_of_enemies=1,
+            )
+        ).get()
+
+        for student in students:
+            num_friends = len(
+                [
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.FRIEND
+                ]
+            )
+            num_enemies = len(
+                [
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.ENEMY
+                ]
+            )
+
+            self.assertEqual(4, num_friends)
+            self.assertEqual(1, num_enemies)
+
+    def test_get__each_student_has_correct_num_friends_and_enemies_non_divisible_class_size(
+        self,
+    ):
+        students = MockStudentProvider(
+            MockStudentProviderSettings(
+                number_of_students=10,
+                number_of_friends=3,
+                number_of_enemies=1,
+            )
+        ).get()
+
+        for student in students:
+            num_friends = len(
+                [
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.FRIEND
+                ]
+            )
+            num_enemies = len(
+                [
+                    1
+                    for stu_id, relation in student.relationships.items()
+                    if relation == Relationship.ENEMY
+                ]
+            )
+
+            self.assertEqual(3, num_friends)
+            self.assertEqual(1, num_enemies)
+
+    def test_get__class_size_too_small(self):
```

Comment:
I'm being a bit greedy here, but you've added a few new things to care about that should be tested (I believe):
- a student's friend cannot also be their enemy, if there are enough people
- if cluster is used, we should check the cluster is a cluster